backends/dpdk/dpdkArch.cpp

```diff
@@ -2448,4 +2448,111 @@ const IR::Node* ElimHeaderCopy::postorder(IR::Member* m) {
+    } else if (k->is<IR::Member>()) {  // handle SerEnum members
+            auto mem = k->to<IR::Member>();
+            auto se = mem->type->to<IR::Type_SerEnum>();
+            auto ei = P4::EnumInstance::resolve(mem, typeMap);
+            if (!ei) return -1;
+            if (auto sei = ei->to<P4::SerEnumInstance>()) {
+                auto type = sei->value->to<IR::Constant>();
+                auto w = se->type->width_bits();
+                BUG_CHECK(w == keyWidth, "SerEnum bitwidth mismatch");
+                return type->value;
+            }
+            ::error(ErrorType::ERR_INVALID, "%1% invalid Member key expression", k);
+            return -1;
+    } else if (k->is<IR::Cast>()) {
+        return convertSimpleKeyExpressionToBigInt(k->to<IR::Cast>()->expr, keyWidth, typeMap);
+    } else {
+        ::error(ErrorType::ERR_INVALID, "%1% invalid key expression", k);
+        return -1;
+    }
+}
+
+void EmitDpdkTableConfig::addAction(const IR::Expression* actionRef,
+                P4::ReferenceMap* refMap,
+                P4::TypeMap* typeMap) {
+    auto actionCall = actionRef->to<IR::MethodCallExpression>();
+    auto method = actionCall->method->to<IR::PathExpression>()->path;
+    auto decl = refMap->getDeclaration(method, true);
+    auto actionDecl = decl->to<IR::P4Action>();
+    auto actionName = actionDecl->controlPlaneName();
+    dpdkTableConfigFile<<actionName<<" ";
+    std::vector<cstring> paramNames;
+    std::vector<big_int> argVals;
+    auto parameter = actionDecl->parameters->parameters.at(0);
+    auto type = typeMap->getType(parameter);
+    if (auto actArg = type->to<IR::Type_Struct>()) {
+        for (auto f : actArg->fields)
+            paramNames.push_back(f->name);
+    }
+    for (auto args : *actionCall->arguments) {
+        for (auto arg : args->expression->to<IR::ListExpression>()->components) {
+            auto ei = P4::EnumInstance::resolve(arg, typeMap);
+            if (arg->is<IR::Constant>()) {
+                auto constant = arg->to<IR::Constant>();
+                auto argValue = constant->value;
+                argVals.push_back(argValue);
+            } else if (arg->is<IR::BoolLiteral>()) {
+                auto constant = arg->to<IR::BoolLiteral>();
+                auto argValue = static_cast<big_int>(constant->value ? 1 : 0);
+                argVals.push_back(argValue);
+            } else if (ei != nullptr && ei->is<P4::SerEnumInstance>()) {
+                auto sei = ei->to<P4::SerEnumInstance>();
+                auto argValue = sei->value->to<IR::Constant>();
+                argVals.push_back(argValue->value);
+            } else {
+                ::error(ErrorType::ERR_UNSUPPORTED,
+                        "%1% unsupported argument expression", arg);
+                continue;
+            }
+        }
+    }
+
+    for (size_t i = 0; i < argVals.size(); i++) {
+        dpdkTableConfigFile<<paramNames[i] << " "<< argVals[i]<<" ";
```

Comment:
these print statements can be made as utility function, just taking a string as input. It will look cleaner

Reply:
ok